docs/user/agent/add-nodes.md

```diff
@@ -0,0 +1,97 @@
+# Adding a node via the node-joiner tool
+
+## Pre-requisites
+1. The `oc` tool must be available in the execution environment (the "user host").
+2. The user host has a valid network connection to the target OpenShift cluster to be expanded.
+3. The user host has a valid pull-secret.
+
+## Setup
+1. Download the [node-joiner.sh](./node-joiner.sh) script in a working directory in
+   the user host (the "assets folder").
+2. Create a `nodes-config.yaml` in the assets folder. This configuration file must contain the 
+   list of all the nodes that the user wants to add to the target cluster. For each node it must be
+   specified at least the name and the primary interface mac address, for example:
+```
+hosts:
+    - hostname: extra-worker-0
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:7c
+    - hostname: extra-worker-1
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:8c
+    - hostname: extra-worker-2
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:9c
+```
+3. Optionally, it's possible to specify - for each node - an `NMState` configuration block
+   (it will be applied during the first boot), for example:
+```
+hosts:
+    - hostname: extra-worker-0
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:7c
+      networkConfig:
+        interfaces:
+          - name: eth0
+            type: ethernet
+            state: up
+            mac-address: 00:02:46:e3:9e:7c
+            ipv4:
+              enabled: true
+              address:
+                - ip: 192.168.111.90
+                  prefix-length: 24
+              dhcp: false
+        dns-resolver:
+          config:
+            server:
+              - 192.168.111.1
+        routes:
+          config:
+            - destination: 0.0.0.0/0 
+              next-hop-address: 192.168.111.1
+              next-hop-interface: eth0
+              table-id: 254
+    - hostname: extra-worker-1
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:8c
+    - hostname: extra-worker-2
+      interfaces:
+        - name: eth0
+          macAddress: 00:02:46:e3:9e:9c
+
+## ISO generation
```

Comment:
Just a note about using the script in the appliance model, will it be possible to incorporate it as part of the config-image API? I.e. for the appliance, we should use the same ISO for every machine. Thus, I wonder if we could add some property to the config-image that triggers the day2 flow. So we could use a single ABI image for both day1 and day2.